src/module/actions/saveAction.js

```diff
@@ -1,37 +1,59 @@
 import { diff } from 'deep-object-diff'
 import { processResponseData } from '../helpers/processResponseData'
 import { getExpectedResponse } from '../helpers/getExpectedResponse'
+import { saveOptions as applySaveOptions } from './options/saveOptions'
 
 /**
  * Update an existing resource
  *
  * @param {ResourcefulApi} api
+ * @param {Boolean} isCollection
  * @param {String} moduleName
+ * @param {Object} defaultQuery
  */
-export function saveAction (api, isCollection, moduleName) {
+export function saveAction (api, isCollection, moduleName, defaultQuery = {}) {
   return new Proxy(() => {}, {
-    apply (target, thisArg, [vuexFns, id]) {
+    apply (target, thisArg, [vuexFns, params]) {
+      const id = (typeof params === 'object') ? params.id : params
+      const options = Object.prototype.hasOwnProperty.call(params, 'options') ? params.options : null
+
       if (typeof id === 'undefined') {
         throw new Error('You must pass an object id to this action')
       }
 
-      const currentItemState = JSON.parse(JSON.stringify((isCollection)
-        ? thisArg.state[moduleName].items[id]
-        : thisArg.state[moduleName].item))
+      let currentItemState = null
+      let initialItemState = null
+      if (Object.prototype.hasOwnProperty.call(defaultQuery, 'group')) {
+        currentItemState = JSON.parse(JSON.stringify((isCollection)
+          ? thisArg.state[moduleName][defaultQuery.group].items[id]
+          : thisArg.state[moduleName][defaultQuery.group].item))
+
+        initialItemState = JSON.parse(JSON.stringify((isCollection)
+          ? thisArg.state[moduleName][defaultQuery.group].initial[id]
+          : thisArg.state[moduleName][defaultQuery.group].initial))
+      } else {
+        currentItemState = JSON.parse(JSON.stringify((isCollection)
+          ? thisArg.state[moduleName].items[id]
+          : thisArg.state[moduleName].item))
 
-      const initialItemState = JSON.parse(JSON.stringify((isCollection)
-        ? thisArg.state[moduleName].initial[id]
-        : thisArg.state[moduleName].initial))
+        initialItemState = JSON.parse(JSON.stringify((isCollection)
+          ? thisArg.state[moduleName].initial[id]
+          : thisArg.state[moduleName].initial))
+      }
 
-      const changedItemState = diff(initialItemState, currentItemState)
+      let changedItemState = diff(initialItemState, currentItemState)
 
       if (Object.prototype.hasOwnProperty.call(changedItemState, 'relationships')) {
         for (const relationship in changedItemState.relationships) {
           changedItemState.relationships[relationship] = currentItemState.relationships[relationship]
         }
       }
 
-      vuexFns.commit('startLoading', null)
+      if (options) {
+        changedItemState = applySaveOptions({ currentItemState, changedItemState, options })
+      }
```

Comment:
I feel like this and line 44 should be one attributes-to-be-sent resolver. Not exactly sure on how to approach this, however.

Reply:
The Problem here is, that we first have to create the diff, then add the relationships and after all apply the options. Or am I wrong?